shlr/gdb/src/gdbclient/xml.c

```diff
@@ -554,6 +554,9 @@ static int _resolve_arch(libgdbr_t *g, char *xml_data) {
 		if (strstr (xml_data, "com.apple.debugserver.arm64")) {
 			g->target.arch = RZ_SYS_ARCH_ARM;
 			g->target.bits = 64;
+		} else if (strstr (xml_data, "org.gnu.gdb.riscv")) {
+			g->target.arch = RZ_SYS_ARCH_RISCV;
+			g->target.bits = 64;
```

Comment:
It seems that gdb doesn't provide any information in the headers to differentiate 32bit and 64bit riscv, both have `org.gnu.gdb.riscv` and `<architecture>riscv</architecture>`. We should determine the `g->target.bits` based on the bitsize of "pc".

64:
https://github.com/bminor/binutils-gdb/blob/1fa1262d5a865a0f52c381f6465a3965e95c7904/gdb/testsuite/gdb.arch/riscv-tdesc-loading-02.xml
32:
https://github.com/bminor/binutils-gdb/blob/1fa1262d5a865a0f52c381f6465a3965e95c7904/gdb/testsuite/gdb.arch/riscv-tdesc-loading-04.xml

Reply:
Probably is better to fix this with another PR